src/lib/rule2F.ts

```diff
@@ -0,0 +1,226 @@
+import {Context, getDefaultContext} from './context';
+import {computeTextAlternative} from './compute_text_alternative';
+
+// Explicit roles allowing 'name from content'
+// (https://www.w3.org/TR/wai-aria/#namefromcontent)
+// *Spec version 1.1
+const NAME_FROM_CONTENT_ROLES = [
+  'button',
+  'cell',
+  'checkbox',
+  'columnheader',
+  'gridcell',
+  'heading',
+  'link',
+  'menuitem',
+  'menuitemcheckbox',
+  'menuitemradio',
+  'option',
+  'radio',
+  'row',
+  'rowgroup',
+  'rowheader',
+  'switch',
+  'tab',
+  'tooltip',
+  'tree',
+  'treeitem',
+];
+
+/**
+ * HTML element types that allow name from content according
+ * to their implicit aria roles.
+ * (https://www.w3.org/TR/html-aam-1.0/#html-element-role-mappings)
+ */
+const NAME_FROM_CONTENT_ELEM_NODE_NAME = [
+  'h1',
+  'h2',
+  'h3',
+  'h4',
+  'h5',
+  'h6',
+  'tbody',
+  'thead',
+  'tfoot',
+  'summary'
+];
+
+/**
+ * HTML input elements allow name from content when
+ * they have the following values for 'type' attribute
+ * (https://www.w3.org/TR/html-aam-1.0/#html-element-role-mappings)
+ */
+const NAME_FROM_CONTENT_INPUT_TYPES = [
+  'button',
+  'checkbox',
+  'image',
+  'radio',
+  'reset',
+  'submit',
+];
+
+/**
+ * Some HTML elements allow name from context only if certain
+ * conditions apply. This maps element types to functions that
+ * determine if a specific element of that type allows name from content
+ * (https://www.w3.org/TR/html-aam-1.0/#html-element-role-mappings)
+ */
+const NAME_FROM_CONTENT_FUNCTION_OF_ELEM: {[key: string]: Function} = {
```

Comment:
You can be more specific with the type of `Function`, it's `{[key: string]: (elem: HTMLElement) => boolean}`.